src/common-ui/Text.tsx

```diff
@@ -11,31 +11,35 @@ type HeaderText = {
 };
 
 type SpanText = {
-  type: 'span';
+  type: 'content';
   tag: 'span';
 };
 
 type LiText = {
-  type: 'div';
+  type: 'content';
   tag: 'div';
 };
 
 type PText = {
-  type: 'p';
+  type: 'content';
   tag: 'p';
 };
 
+export type HeaderType = HeaderText | SpanText | LiText | PText;
+
 interface TextProps extends StrictPropsWithChildren {
```

Comment:
```suggestion
type TextProps = {
  type: HeaderType;
  weight?: 'bold' | 'regular';
  fontSize?: number;
  color?: keyof typeof theme.colors;
} & HTMLAttributes<HTMLSpanElement | HTMLHeadElement | HTMLParagraphElement>;
```
Text 컴포넌트의 props 를 위와 같이 html tag attributes 들도 포함하도록 확장하면 따로 customCss 를 받지 않아도 될것같습니답!! (style, className 이 props에 포함되기 때문에)

또 Text 컴포넌트에 바로 onClick 과 같은 이벤트핸들러도 붙일 수 있어서 사용하기 좋을것같아용

Reply:
네이밍 부터 요상했네요 TextProps에 HeaderProps라니 😑
말씀해주신 대로 처리하는게 좀 더 사용성이 좋을 것 같아요!
수정해보겠습니다 🫡